xesmf/frontend.py

```diff
@@ -54,8 +54,14 @@ def ds_to_ESMFgrid(ds, need_bounds=False, periodic=None, append=None):
     lat = np.asarray(ds['lat'])
     lon, lat = as_2d_mesh(lon, lat)
 
+    if 'mask' in ds:
+        mask = np.asarray(ds['mask'])
+        print(mask.shape)
+    else:
+        mask = None
+
     # tranpose the arrays so they become Fortran-ordered
-    grid = esmf_grid(lon.T, lat.T, periodic=periodic)
+    grid = esmf_grid(lon.T, lat.T, periodic=periodic, mask=mask)
```

Comment:
Would it be clearer to have `mask.T` as well insted of transposing it inside `esgf_grid` ?